totalRP3/core/impl/profiles.lua

```diff
@@ -223,10 +225,21 @@ end
 -- Refresh list display
 local function uiInitProfileList()
 	wipe(profileListID);
+	local profileSearch = Utils.str.emptyToNil(TRP3_ProfileManagerSearch:GetText());
 	for profileID, _ in pairs(profiles) do
-		tinsert(profileListID, profileID);
+		if not profileSearch or safeMatch(profiles[profileID].profileName:lower(), profileSearch:lower()) then
+			tinsert(profileListID, profileID);
+		end
+	end
+
+	local size = tsize(profileListID);
```

Comment:
`profileListID` is being used as an array-like table as evidenced by the `tinsert` call - so `tsize` could be changed to `#` and be logarithmic time instead of function-call-overhead-and-linear-time.